prometheus_client/utils.py

```diff
@@ -1,4 +1,5 @@
 import math
+from threading import _PyRLock  # type: ignore[attr-defined]
```

Comment:
why not from threading import RLock?

Reply:
`RLock` does not have a `locked()` method.

https://docs.python.org/3/library/threading.html#rlock-objects

However, `_PyRLock` provides `_count` which I am using to provide that method to satisfy the rest of the code's usage of the lock.

Reply:
Look closely at `threading`, `RLock` is a factory function that either returns an instance of `_CRLock` or `_PyRLock`. The former does not have a `_count` attribute though the latter does. I am choosing `_PyRLock` explicitly to implement `locked()`.

Reply:
nice! Sorry I missed that!

Reply:
No problem. I had to go re-read the `threading` module source to jog my memory.

Reply:
Unless I am missing something only tests depend on `locked()`. Could we instead use `RLock` in the code, and change the test to check the lock in a different manner? E.g. trying to acquire the lock with blocked=false and testing if we acquired the lock or not.

Reply:
Or maybe more easily just derive the class from `RLock` and implement `locked()` method using `blocked=False` as you suggest.

Reply:
Well, nevermind, as I mentioned above, `RLock` is a factory function (can't derive). Could instead have a `locked(lock) -> bool` function used in tests.

Reply:
That seems reasonable to me, it's only one place in tests so we don't need anything too fancy.